src/logic/logic.js

```diff
@@ -0,0 +1,159 @@
+/**
+ Utilities for working with program bytes.
+ */
+
+const langspec = require("./langspec.json")
```

Comment:
We should probably checksum this file and compare upon loading to prevent erroneous changes to the file.

Reply:
I think it is not a big deal - client side checking mainly aims reducing the amount of invalid programs send to the network.